From 833dbfa1622ebf431f9827468a70a77eb1ea22f3 Mon Sep 17 00:00:00 2001
From: Mahesh R Vaidya <mahesh.r.vaidya@intel.com>
Date: Thu, 8 May 2025 02:40:43 +0530
Subject: Config-3 SinglePhy 1x2.5G Device tree changes

        1. Created MRPHY gmii-to-sgmii adapter node in tsn cfg3 dts file.
        2. Modified the gmac0 node in dts file to support gmii mac mode and
           phy mode to 2500base-x requried for TSN CFG-3.

Upstream-Status: Inappropriate [Changes only for TSN CFG3 system design]
Signed-off-by: Mahesh R Vaidya <mahesh.r.vaidya@intel.com>
---
 arch/arm64/boot/dts/intel/Makefile            |  1 +
 .../intel/socfpga_agilex5_socdk_tsn_cfg3.dts  | 43 +++++++++++++++++++
 2 files changed, 44 insertions(+)
 create mode 100644 arch/arm64/boot/dts/intel/socfpga_agilex5_socdk_tsn_cfg3.dts

diff --git a/arch/arm64/boot/dts/intel/Makefile b/arch/arm64/boot/dts/intel/Makefile
index 1f62d0e4f3db..d7b40080f702 100644
--- a/arch/arm64/boot/dts/intel/Makefile
+++ b/arch/arm64/boot/dts/intel/Makefile
@@ -15,6 +15,7 @@ dtb-$(CONFIG_ARCH_INTEL_SOCFPGA) += socfpga_agilex_n6000.dtb \
 				socfpga_agilex5_socdk_swvp_b0.dtb \
 				socfpga_agilex5_socdk_tsn_cfg2.dtb \
 				socfpga_agilex5_socdk_tsn_cfg2_b0.dtb \
+				socfpga_agilex5_socdk_tsn_cfg3.dtb \
 				socfpga_agilex5_socdk_modular.dtb \
 				socfpga_agilex7m_socdk.dtb \
 				socfpga_agilex7f_socdk_pcie_root_port.dtb \
diff --git a/arch/arm64/boot/dts/intel/socfpga_agilex5_socdk_tsn_cfg3.dts b/arch/arm64/boot/dts/intel/socfpga_agilex5_socdk_tsn_cfg3.dts
new file mode 100644
index 000000000000..38f65fc4c787
--- /dev/null
+++ b/arch/arm64/boot/dts/intel/socfpga_agilex5_socdk_tsn_cfg3.dts
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright (c) 2024, Intel Corporation
+ */
+#include "socfpga_agilex5_socdk.dts"
+
+/ {
+	gmii_to_sgmii_converter: phy@30020100 {
+		compatible = "altr,gmii-to-sgmii-2.0";
+		reg = <0x0 0x30020100 0x0 0x80>;
+		reg-names = "mrphy";
+	};
+};
+
+&gmac0 {
+	status = "okay";
+	phy-mode = "2500base-x";
+	mac-mode = "gmii";
+	phy-handle = <&emac0_phy0>;
+	max-speed = <2500>;
+	max-frame-size = <9000>;
+	#address-cells = <1>;
+	#size-cells = <1>;
+	altr,gmii-to-sgmii-converter = <&gmii_to_sgmii_converter>;
+
+	mdio0 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "snps,dwmac-mdio";
+		emac0_phy0: ethernet-phy@0 {
+			reg = <0>;
+		};
+	};
+};
+
+&gmac1 {
+        status = "disabled";
+};
+
+&gmac2 {
+	status = "disabled";
+};
+
-- 
2.34.1

